substrate/frame/identity/src/types.rs

```diff
@@ -430,7 +294,52 @@ pub struct RegistrarInfo<
 
 	/// Relevant fields for this registrar. Registrar judgements are limited to attestations on
 	/// these fields.
-	pub fields: IdentityFields,
+	pub fields: IdentityFields<IdField>,
+}
+
+/// Wrapper type for `BitFlags<IdentityField>` that implements `Codec`.
+#[derive(Clone, Copy, PartialEq, RuntimeDebug)]
+pub struct IdentityFields<IdField: BitFlag>(pub BitFlags<IdField>);
+
+impl<IdField: U64BitFlag> Default for IdentityFields<IdField> {
+	fn default() -> Self {
+		Self(Default::default())
+	}
+}
+
+impl<IdField: U64BitFlag> MaxEncodedLen for IdentityFields<IdField>
```

Comment:
You can remove the code below and use the `impl_codec_bitflags` macro from [there](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/nfts/src/macros.rs#L29)
P. S. Previously we had a different `enumflags2` version for the identity pallet and nfts, but since we synced the versions now, we could extract that macro into a common folder and re-use the code

Reply:
Unfortunately it doesn't work because of the generics, but I don't think it's too ugly either way.

Reply:
hmm.. is it possible to avoid that generic? I think it could work without it, but maybe I'm missing smth